src/runtime-support.cc

```diff
@@ -303,7 +303,7 @@ void rt::protocol_end() {
         c->analyzer->RemoveChildAnalyzer(i);
 }
 
-inline rt::cookie::FileState* _file_state(rt::Cookie* cookie) {
+inline rt::cookie::FileStateStack* _file_state_stack(rt::Cookie* cookie) {
```

Comment:
This function and `_file_state` can never return a nullptr which I had to remind myself of while reviewing the call sites. How do you feel about making them all return refs instead of pointers, even though the return value is mutable?

Reply:
I don't like returning references to mutable values, hurts readability and doesn't match the style of the code elsewhere.